src/rvoice/fluid_rev.c

```diff
@@ -481,11 +481,31 @@ fluid_revmodel_processmix(fluid_revmodel_t* rev, fluid_real_t *in,
 }
 
 static void
+/* scalelevelwidth is compensation weigth factor to get output
+   amplitude rather  independent of width setting.
+   0: the output amplitude is fully dependant of width setting.
+   >0: the output amplitude is less dependant of width setting.
+   With a scalelevelwidthof of 0.2 the output amplitude is 
+   rather independent of width setting.
+*/
+#define scalelevelwidth 0.2f 
 fluid_revmodel_update(fluid_revmodel_t* rev)
 {
   /* Recalculate internal values after parameter change */
   int i;
-
+  
+  /* The stereo amplitude equation  (wet1 and wet1 below) have a 
+  tendency to produce high amplitude with high width value ( 1 < width < 100).
+  This result in a unwanted output clipped by the audio card.
+  To avoid this dependency, we divide by (1 + rev->width * scalelevelwidth)
+  Actually, with a scalelevelwidthof 0.2, (regardless of level setting), 
+  the output amplitude (wet) seems rather independent of width setting */	
+  rev->wet = (rev->level * scalewet) / 
+             (1.0f + rev->width * scalelevelwidth);
```

Comment:
Looks like `rev->wet` is only use in the local scope here. It should be removed from `struct _fluid_revmodel_t` then.

Reply:
Yes. Since the arrival of rev->level,   rev->wet need to be removed.